### PR TITLE
LPS-85856 mail-reader-service: Integrate Message into new indexer model

### PR DESCRIPTION
**Pull request by brandizzi**

### CODE REVIEW

**Comment by foshiro-zz** [comment]
ci:test:sf

**Comment by foshiro-zz** [comment]
ci:test:search

File: modules/apps/mail-reader/mail-reader-test/src/testIntegration/java/com/liferay/mail/reader/search/test/MailReaderIndexerReindexTest.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.search.Document;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.rule.SynchronousDestinationTestRule;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.search.test.util.IndexerFixture;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerTestRule;
```
Comment (foshiro-zz) [inline modules/apps/mail-reader/mail-reader-test/src/testIntegration/java/com/liferay/mail/reader/search/test/MailReaderIndexerReindexTest.java]: Please, change to import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;

File: modules/apps/mail-reader/mail-reader-test/src/testIntegration/java/com/liferay/mail/reader/search/test/MailReaderIndexerReindexTest.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+ */
+@RunWith(Arquillian.class)
+public class MailReaderIndexerReindexTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerTestRule.INSTANCE,
```
Comment (foshiro-zz) [inline modules/apps/mail-reader/mail-reader-test/src/testIntegration/java/com/liferay/mail/reader/search/test/MailReaderIndexerReindexTest.java]: Please, change to PermissionCheckerMethodTestRule.INSTANCE

File: modules/apps/mail-reader/mail-reader-test/src/testIntegration/java/com/liferay/mail/reader/search/test/MessageIndexerIndexedFieldsTest.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.rule.SynchronousDestinationTestRule;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.search.test.util.FieldValuesAssert;
+import com.liferay.portal.search.test.util.IndexedFieldsFixture;
+import com.liferay.portal.search.test.util.IndexerFixture;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerTestRule;
```
Comment (foshiro-zz) [inline modules/apps/mail-reader/mail-reader-test/src/testIntegration/java/com/liferay/mail/reader/search/test/MessageIndexerIndexedFieldsTest.java]: Please, change to import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;

File: modules/apps/mail-reader/mail-reader-test/src/testIntegration/java/com/liferay/mail/reader/search/test/MessageIndexerIndexedFieldsTest.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+ */
+@RunWith(Arquillian.class)
+public class MessageIndexerIndexedFieldsTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerTestRule.INSTANCE,
```
Comment (foshiro-zz) [inline modules/apps/mail-reader/mail-reader-test/src/testIntegration/java/com/liferay/mail/reader/search/test/MessageIndexerIndexedFieldsTest.java]: Please, change to PermissionCheckerMethodTestRule.INSTANCE

File: modules/apps/mail-reader/mail-reader-test/build.gradle [hunk 5]
```diff
@@ -0,0 +1,12 @@
+dependencies {
+	testIntegrationCompile group: "com.liferay.portal", name: "com.liferay.portal.impl", version: "default"
+	testIntegrationCompile group: "com.liferay.portal", name: "com.liferay.portal.kernel", version: "default"
+	testIntegrationCompile group: "org.osgi", name: "org.osgi.core", version: "5.0.0"
```
Comment (foshiro-zz) [inline modules/apps/mail-reader/mail-reader-test/build.gradle]: Please, change to use newer version: 6.0.0

File: modules/apps/mail-reader/mail-reader-service/src/main/java/com/liferay/mail/reader/internal/search/spi/model/index/contributor/AccountModelDocumentContributor.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	service = ModelDocumentContributor.class
+)
+public class AccountModelDocumentContributor
+	implements ModelDocumentContributor<Account> {
+
+	@Override
+	public void contribute(Document document, Account account) {
+		ExpandoBridge expandoBridge = account.getExpandoBridge();
+
+		document.addKeyword("accountId", account.getAccountId());
```
Comment (foshiro-zz) [inline modules/apps/mail-reader/mail-reader-service/src/main/java/com/liferay/mail/reader/internal/search/spi/model/index/contributor/AccountModelDocumentContributor.java:43]: Please, create a new constant FIELD.ACCOUNT_ID and use it here

File: modules/apps/mail-reader/mail-reader-service/src/main/java/com/liferay/mail/reader/internal/search/spi/model/index/contributor/FolderModelDocumentContributor.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+	implements ModelDocumentContributor<Folder> {
+
+	@Override
+	public void contribute(Document document, Folder folder) {
+		ExpandoBridge expandoBridge = folder.getExpandoBridge();
+
+		document.addKeyword(Field.FOLDER_ID, folder.getFolderId());
+		document.addText(Field.NAME, folder.getDisplayName());
+
+		document.addKeyword("accountId", folder.getAccountId());
```
Comment (foshiro-zz) [inline modules/apps/mail-reader/mail-reader-service/src/main/java/com/liferay/mail/reader/internal/search/spi/model/index/contributor/FolderModelDocumentContributor.java:46]: Please, create a new constant FIELD.ACCOUNT_ID and use it here

File: modules/apps/mail-reader/mail-reader-service/src/main/java/com/liferay/mail/reader/internal/search/spi/model/index/contributor/MessageModelDocumentContributor.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+	@Override
+	public void contribute(Document document, Message message) {
+		ExpandoBridge expandoBridge = message.getExpandoBridge();
+
+		document.addText(
+			Field.CONTENT, HtmlUtil.extractText(message.getBody()));
+		document.addKeyword(Field.FOLDER_ID, message.getFolderId());
+		document.addText(Field.TITLE, message.getSubject());
+
+		document.addKeyword("accountId", message.getAccountId());
```
Comment (foshiro-zz) [inline modules/apps/mail-reader/mail-reader-service/src/main/java/com/liferay/mail/reader/internal/search/spi/model/index/contributor/MessageModelDocumentContributor.java:49]: Please, create a new constant FIELD.ACCOUNT_ID and use it here

File: modules/apps/mail-reader/mail-reader-test/src/testIntegration/java/com/liferay/mail/reader/search/test/MessageIndexerIndexedFieldsTest.java [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+
+		map.put(Field.FOLDER_ID, String.valueOf(message.getFolderId()));
+
+		map.put(Field.TITLE, message.getSubject());
+
+		map.put(Field.USER_ID, String.valueOf(message.getUserId()));
+
+		map.put(Field.USER_NAME, StringUtil.lowerCase(message.getUserName()));
+
+		map.put("accountId", String.valueOf(message.getAccountId()));
```
Comment (foshiro-zz) [inline modules/apps/mail-reader/mail-reader-test/src/testIntegration/java/com/liferay/mail/reader/search/test/MessageIndexerIndexedFieldsTest.java:154]: Please, create a new constant FIELD.ACCOUNT_ID and use it here

File: modules/apps/mail-reader/mail-reader-service/src/main/java/com/liferay/mail/reader/internal/search/spi/model/index/contributor/MessageModelDocumentContributor.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+		document.addKeyword("remoteMessageId", message.getRemoteMessageId());
```
Comment (foshiro-zz) [inline modules/apps/mail-reader/mail-reader-service/src/main/java/com/liferay/mail/reader/internal/search/spi/model/index/contributor/MessageModelDocumentContributor.java:50]: Please, create a new constant FIELD.REMOTE_MESSAGE_ID and use it here

File: modules/apps/mail-reader/mail-reader-test/src/testIntegration/java/com/liferay/mail/reader/search/test/MessageIndexerIndexedFieldsTest.java [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+
+		map.put(
+			"remoteMessageId", String.valueOf(message.getRemoteMessageId()));
```
Comment (foshiro-zz) [inline modules/apps/mail-reader/mail-reader-test/src/testIntegration/java/com/liferay/mail/reader/search/test/MessageIndexerIndexedFieldsTest.java:157]: Please, create a new constant FIELD.REMOTE_MESSAGE_ID and use it here

**Review by foshiro-zz — CHANGES_REQUESTED** [review]
@luanmaoski, please check my comments for change request. If you have any question, please let me know.

**Comment by vagnerbc** [comment]
Changes updated

**Comment by foshiro-zz** [comment]
Thanks @vagnerbc 
LGTM!

**Review by foshiro-zz — APPROVED** [review]

**Comment by foshiro-zz** [comment]
ci:test:sf

**Comment by foshiro-zz** [comment]
ci:test:search

**Comment by foshiro-zz** [comment]
ci:test:sf

**Comment by foshiro-zz** [comment]
ci:test:sf

**Comment by foshiro-zz** [comment]
ci:test:search

**Comment by foshiro-zz** [comment]
Submitted to https://github.com/brandizzi/liferay-portal/pull/778

**Pull request closed without merging**